### PR TITLE
Handle ingress failures

### PR DESCRIPTION
**Pull request by Ladas**
Log errors nicely and wait before repating the refresh, to
avoin bombarding the server when there is an error.

### CODE REVIEW

File: lib/topological_inventory/amazon/collector.rb [hunk 1]
```diff
@@ -114,6 +122,9 @@ def save_inventory(collections, refresh_state_uuid = nil, refresh_state_part_uui
             :refresh_state_part_uuid => refresh_state_part_uuid,
           )
         )
+      rescue => e
+        logger.error("Error when sending payload to Ingress API. Error message: #{e.message}. Error headers: #{e&.response_headers}")
+        raise e
```
Comment (agrare) [inline lib/topological_inventory/amazon/collector.rb]: You can just `raise` no need to `raise e`, it'll reraise the last exception for you

File: lib/topological_inventory/amazon/collector.rb [hunk 2]
```diff
@@ -128,6 +139,9 @@ def sweep_inventory(refresh_state_uuid, total_parts, sweep_scope)
             :sweep_scope        => sweep_scope,
           )
         )
+      rescue => e
+        logger.error("Error when sending payload to Ingress API. Error message: #{e.message}. Error headers: #{e&.response_headers}")
+        raise e
```
Comment (agrare) [inline lib/topological_inventory/amazon/collector.rb]: Same

File: lib/topological_inventory/amazon/collector.rb [hunk 3]
```diff
@@ -31,8 +31,16 @@ def initialize(source, access_key_id, secret_access_key, batch_size: 1_000, poll
       end
 
       def collect!
-        entity_types.each do |entity_type|
-          process_entity(entity_type)
+        loop do
+          begin
```
Comment (agrare) [inline lib/topological_inventory/amazon/collector.rb]: Hm don't think you need the begin, rescue at the same level as the loop will work and save some indentation

Comment (Ladas) [reply lib/topological_inventory/amazon/collector.rb]: the is for ruby 2.5, so it won't pass the CI

File: lib/topological_inventory/amazon/collector.rb (hunk 1 above)
Comment (Ladas) [reply lib/topological_inventory/amazon/collector.rb]: done

File: lib/topological_inventory/amazon/collector.rb (hunk 2 above)
Comment (Ladas) [reply lib/topological_inventory/amazon/collector.rb]: done

File: lib/topological_inventory/amazon/collector.rb (hunk 3 above)
Comment (agrare) [reply lib/topological_inventory/amazon/collector.rb]: :+1:

File: lib/topological_inventory/amazon/collector.rb [hunk 4]
```diff
@@ -114,6 +122,9 @@ def save_inventory(collections, refresh_state_uuid = nil, refresh_state_part_uui
             :refresh_state_part_uuid => refresh_state_part_uuid,
           )
         )
+      rescue => e
+        logger.error("Error when sending payload to Ingress API. Error message: #{e.message}. Error headers: #{e&.response_headers}")
```
Comment (agrare) [inline lib/topological_inventory/amazon/collector.rb]: `e&.response_headers`

The save navigator doesn't handle method_missing:
```
>> begin; raise; rescue => e; puts e&.response_headers; end
Traceback (most recent call last):
        3: from /usr/bin/irb:11:in `<main>'
        2: from (irb):6
        1: from (irb):6:in `rescue in irb_binding'
NoMethodError (undefined method `response_headers' for RuntimeError:RuntimeError)
```

Comment (Ladas) [reply lib/topological_inventory/amazon/collector.rb]: ok, so I should limit this on the API error only

Comment (agrare) [reply lib/topological_inventory/amazon/collector.rb]: Yeah or either `.try()` or `if e.respond_to?()` would work

**Pull request merged**